### ISSUE TITLE
2022-02-17 nightly fails to build

### ISSUE THREAD

**Issue by msfjarvis** [oxalica/rust-overlay#73]
Using the default profile the autoPatchelfHook fails due to missing libstdc++6 in nixpkgs, I can get a build log in some time if required.

**Comment by chvp** [comment]
```
@nix { "action": "setPhase", "phase": "unpackPhase" }
unpacking sources
unpacking source archive /nix/store/5ypsxbhfwqcd18j65sv0w7j2q8nc08b3-rustc-nightly-x86_64-unknown-linux-gnu.tar.xz
source root is rustc-nightly-x86_64-unknown-linux-gnu
setting SOURCE_DATE_EPOCH to timestamp 1645042130 of file rustc-nightly-x86_64-unknown-linux-gnu/version
@nix { "action": "setPhase", "phase": "patchPhase" }
patching sources
@nix { "action": "setPhase", "phase": "installPhase" }
installing
Installing component rustc
@nix { "action": "setPhase", "phase": "fixupPhase" }
post-installation fixup
shrinking RPATHs of ELF executables and libraries in /nix/store/gijpr102l7456xmh5p659ifxa8kj4139-rustc-1.60.0-nightly-2022-02-17-x86_64-unknown-linux-gnu
shrinking /nix/store/gijpr102l7456xmh5p659ifxa8kj4139-rustc-1.60.0-nightly-2022-02-17-x86_64-unknown-linux-gnu/lib/libtest-b81f71b2dfdd83d8.so
shrinking /nix/store/gijpr102l7456xmh5p659ifxa8kj4139-rustc-1.60.0-nightly-2022-02-17-x86_64-unknown-linux-gnu/lib/rustlib/x86_64-unknown-linux-gnu/bin/gcc-ld/ld64
shrinking /nix/store/gijpr102l7456xmh5p659ifxa8kj4139-rustc-1.60.0-nightly-2022-02-17-x86_64-unknown-linux-gnu/lib/rustlib/x86_64-unknown-linux-gnu/bin/gcc-ld/ld
shrinking /nix/store/gijpr102l7456xmh5p659ifxa8kj4139-rustc-1.60.0-nightly-2022-02-17-x86_64-unknown-linux-gnu/lib/rustlib/x86_64-unknown-linux-gnu/bin/rust-lld
shrinking /nix/store/gijpr102l7456xmh5p659ifxa8kj4139-rustc-1.60.0-nightly-2022-02-17-x86_64-unknown-linux-gnu/lib/librustc_driver-920796763ff499ad.so
shrinking /nix/store/gijpr102l7456xmh5p659ifxa8kj4139-rustc-1.60.0-nightly-2022-02-17-x86_64-unknown-linux-gnu/lib/libLLVM-13-rust-1.60.0-nightly.so
shrinking /nix/store/gijpr102l7456xmh5p659ifxa8kj4139-rustc-1.60.0-nightly-2022-02-17-x86_64-unknown-linux-gnu/lib/libstd-fad6565a4a24ca7c.so
shrinking /nix/store/gijpr102l7456xmh5p659ifxa8kj4139-rustc-1.60.0-nightly-2022-02-17-x86_64-unknown-linux-gnu/bin/rustdoc
shrinking /nix/store/gijpr102l7456xmh5p659ifxa8kj4139-rustc-1.60.0-nightly-2022-02-17-x86_64-unknown-linux-gnu/bin/rustc
gzipping man pages under /nix/store/gijpr102l7456xmh5p659ifxa8kj4139-rustc-1.60.0-nightly-2022-02-17-x86_64-unknown-linux-gnu/share/man/
patching script interpreter paths in /nix/store/gijpr102l7456xmh5p659ifxa8kj4139-rustc-1.60.0-nightly-2022-02-17-x86_64-unknown-linux-gnu
/nix/store/gijpr102l7456xmh5p659ifxa8kj4139-rustc-1.60.0-nightly-2022-02-17-x86_64-unknown-linux-gnu/bin/rust-lldb: interpreter directive changed from "#!/bin/sh" to "/nix/store/4nmqxajzaf60yjribkgvj5j54x9yvr1r-bash-5.1-p12/bin/sh"
/nix/store/gijpr102l7456xmh5p659ifxa8kj4139-rustc-1.60.0-nightly-2022-02-17-x86_64-unknown-linux-gnu/bin/rust-gdbgui: interpreter directive changed from "#!/bin/sh" to "/nix/store/4nmqxajzaf60yjribkgvj5j54x9yvr1r-bash-5.1-p12/bin/sh"
/nix/store/gijpr102l7456xmh5p659ifxa8kj4139-rustc-1.60.0-nightly-2022-02-17-x86_64-unknown-linux-gnu/bin/rust-gdb: interpreter directive changed from "#!/bin/sh" to "/nix/store/4nmqxajzaf60yjribkgvj5j54x9yvr1r-bash-5.1-p12/bin/sh"
checking for references to /build/ in /nix/store/gijpr102l7456xmh5p659ifxa8kj4139-rustc-1.60.0-nightly-2022-02-17-x86_64-unknown-linux-gnu...
automatically fixing dependencies for ELF files
searching for dependencies of /nix/store/gijpr102l7456xmh5p659ifxa8kj4139-rustc-1.60.0-nightly-2022-02-17-x86_64-unknown-linux-gnu/lib/libtest-b81f71b2dfdd83d8.so
  libstd-fad6565a4a24ca7c.so -> found: /nix/store/gijpr102l7456xmh5p659ifxa8kj4139-rustc-1.60.0-nightly-2022-02-17-x86_64-unknown-linux-gnu/lib/libstd-fad6565a4a24ca7c.so
setting RPATH to: /nix/store/gijpr102l7456xmh5p659ifxa8kj4139-rustc-1.60.0-nightly-2022-02-17-x86_64-unknown-linux-gnu/lib
searching for dependencies of /nix/store/gijpr102l7456xmh5p659ifxa8kj4139-rustc-1.60.0-nightly-2022-02-17-x86_64-unknown-linux-gnu/lib/rustlib/x86_64-unknown-linux-gnu/bin/gcc-ld/ld64
searching for dependencies of /nix/store/gijpr102l7456xmh5p659ifxa8kj4139-rustc-1.60.0-nightly-2022-02-17-x86_64-unknown-linux-gnu/lib/rustlib/x86_64-unknown-linux-gnu/bin/gcc-ld/ld
searching for dependencies of /nix/store/gijpr102l7456xmh5p659ifxa8kj4139-rustc-1.60.0-nightly-2022-02-17-x86_64-unknown-linux-gnu/lib/rustlib/x86_64-unknown-linux-gnu/bin/rust-lld
  libLLVM-13-rust-1.60.0-nightly.so -> found: /nix/store/gijpr102l7456xmh5p659ifxa8kj4139-rustc-1.60.0-nightly-2022-02-17-x86_64-unknown-linux-gnu/lib/libLLVM-13-rust-1.60.0-nightly.so
  libstdc++.so.6 -> not found!
setting RPATH to: /nix/store/gijpr102l7456xmh5p659ifxa8kj4139-rustc-1.60.0-nightly-2022-02-17-x86_64-unknown-linux-gnu/lib
searching for dependencies of /nix/store/gijpr102l7456xmh5p659ifxa8kj4139-rustc-1.60.0-nightly-2022-02-17-x86_64-unknown-linux-gnu/lib/librustc_driver-920796763ff499ad.so
  libstd-fad6565a4a24ca7c.so -> found: /nix/store/gijpr102l7456xmh5p659ifxa8kj4139-rustc-1.60.0-nightly-2022-02-17-x86_64-unknown-linux-gnu/lib/libstd-fad6565a4a24ca7c.so
  libLLVM-13-rust-1.60.0-nightly.so -> found: /nix/store/gijpr102l7456xmh5p659ifxa8kj4139-rustc-1.60.0-nightly-2022-02-17-x86_64-unknown-linux-gnu/lib/libLLVM-13-rust-1.60.0-nightly.so
setting RPATH to: /nix/store/gijpr102l7456xmh5p659ifxa8kj4139-rustc-1.60.0-nightly-2022-02-17-x86_64-unknown-linux-gnu/lib:/nix/store/gijpr102l7456xmh5p659ifxa8kj4139-rustc-1.60.0-nightly-2022-02-17-x86_64-unknown-linux-gnu/lib
searching for dependencies of /nix/store/gijpr102l7456xmh5p659ifxa8kj4139-rustc-1.60.0-nightly-2022-02-17-x86_64-unknown-linux-gnu/lib/libLLVM-13-rust-1.60.0-nightly.so
  libz.so.1 -> found: /nix/store/bqjh8dc6m5plswxrmmn9x4cpxrrw94lv-zlib-1.2.11/lib/libz.so.1
setting RPATH to: /nix/store/bqjh8dc6m5plswxrmmn9x4cpxrrw94lv-zlib-1.2.11/lib
searching for dependencies of /nix/store/gijpr102l7456xmh5p659ifxa8kj4139-rustc-1.60.0-nightly-2022-02-17-x86_64-unknown-linux-gnu/lib/libstd-fad6565a4a24ca7c.so
searching for dependencies of /nix/store/gijpr102l7456xmh5p659ifxa8kj4139-rustc-1.60.0-nightly-2022-02-17-x86_64-unknown-linux-gnu/bin/rustdoc
  libtest-b81f71b2dfdd83d8.so -> found: /nix/store/gijpr102l7456xmh5p659ifxa8kj4139-rustc-1.60.0-nightly-2022-02-17-x86_64-unknown-linux-gnu/lib/libtest-b81f71b2dfdd83d8.so
  librustc_driver-920796763ff499ad.so -> found: /nix/store/gijpr102l7456xmh5p659ifxa8kj4139-rustc-1.60.0-nightly-2022-02-17-x86_64-unknown-linux-gnu/lib/librustc_driver-920796763ff499ad.so
  libstd-fad6565a4a24ca7c.so -> found: /nix/store/gijpr102l7456xmh5p659ifxa8kj4139-rustc-1.60.0-nightly-2022-02-17-x86_64-unknown-linux-gnu/lib/libstd-fad6565a4a24ca7c.so
setting RPATH to: /nix/store/gijpr102l7456xmh5p659ifxa8kj4139-rustc-1.60.0-nightly-2022-02-17-x86_64-unknown-linux-gnu/lib:/nix/store/gijpr102l7456xmh5p659ifxa8kj4139-rustc-1.60.0-nightly-2022-02-17-x86_64-unknown-linux-gnu/lib:/nix/store/gijpr102l7456xmh5p659ifxa8kj4139-rustc-1.60.0-nightly-2022-02-17-x86_64-unknown-linux-gnu/lib
searching for dependencies of /nix/store/gijpr102l7456xmh5p659ifxa8kj4139-rustc-1.60.0-nightly-2022-02-17-x86_64-unknown-linux-gnu/bin/rustc
  librustc_driver-920796763ff499ad.so -> found: /nix/store/gijpr102l7456xmh5p659ifxa8kj4139-rustc-1.60.0-nightly-2022-02-17-x86_64-unknown-linux-gnu/lib/librustc_driver-920796763ff499ad.so
  libstd-fad6565a4a24ca7c.so -> found: /nix/store/gijpr102l7456xmh5p659ifxa8kj4139-rustc-1.60.0-nightly-2022-02-17-x86_64-unknown-linux-gnu/lib/libstd-fad6565a4a24ca7c.so
setting RPATH to: /nix/store/gijpr102l7456xmh5p659ifxa8kj4139-rustc-1.60.0-nightly-2022-02-17-x86_64-unknown-linux-gnu/lib:/nix/store/gijpr102l7456xmh5p659ifxa8kj4139-rustc-1.60.0-nightly-2022-02-17-x86_64-unknown-linux-gnu/lib
autoPatchelfHook could not satisfy dependency libstdc++.so.6 wanted by /nix/store/gijpr102l7456xmh5p659ifxa8kj4139-rustc-1.60.0-nightly-2022-02-17-x86_64-unknown-linux-gnu/lib/rustlib/x86_64-unknown-linux-gnu/bin/rust-lld
Add the missing dependencies to the build inputs or set autoPatchelfIgnoreMissingDeps=true
```

^ Full build log

**Comment by oxalica** [comment]
FIxed in 8e5bd8e5953d55b4ab8bec29be89cf53ea9f4f27
